dart/lib/src/attachment.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:typed_data';
+export 'attachment_extensions/attachment_extensions.dart';
+
+// https://develop.sentry.dev/sdk/envelopes/#attachment
+
+/// Arbitrary content which gets attached to an event.
+class Attachment {
+  Attachment({
+    required this.content,
+    required this.fileName,
+    AttachmentType? type,
+    String? mimeType,
+  })  : mimeType = mimeType ?? 'application/octet-stream',
+        type = type ?? AttachmentType.attachment;
+
+  /// Creates an [Attachment] from a [Uint8List]
+  factory Attachment.fromUint8List(
+    Uint8List bytes,
+    String fileName, {
+    String? mimeType,
+    AttachmentType? type,
+  }) {
+    return Attachment(
+      type: type,
+      content: bytes,
+      fileName: fileName,
+      mimeType: mimeType,
+    );
+  }
+
+  /// Creates an [Attachment] from a [List<int>]
+  factory Attachment.fromIntList(
+    List<int> bytes,
+    String fileName, {
+    String? mimeType,
+    AttachmentType? type,
+  }) {
+    return Attachment(
+      type: type,
+      content: Uint8List.fromList(bytes),
+      fileName: fileName,
+      mimeType: mimeType,
+    );
+  }
+
+  /// Creates an [Attachment] from [ByteData]
+  factory Attachment.fromByteData(
+    ByteData bytes,
+    String fileName, {
+    String? mimeType,
+    AttachmentType? type,
+  }) {
+    return Attachment(
+      type: type,
+      content: bytes.buffer.asUint8List(),
+      fileName: fileName,
+      mimeType: mimeType,
+    );
+  }
+
+  /// Attachment type.
+  final AttachmentType type;
+
+  /// Attachment content.
+  final Uint8List content;
+
+  /// Attachment file name.
+  final String fileName;
+
+  /// Attachment content type.
+  final String mimeType;
+}
+
+/// Attachment type.
+enum AttachmentType {
```

Comment:
on Java is a String due to being easier to expand types without releasing a new version
maybe @bruno-garcia has opinions, I'm fine as it is as it covers all the cases right now

Reply:
A string probably makes more sense. We already use strings for envelope types, so it would be consistent.

Reply:
On Cocoa and Java, we don't have the `attachmentType` in the API. We just have the content type.

Reply:
we do, it was a bug and I fixed it a while ago
https://github.com/getsentry/sentry-java/blob/47b88ecea6dbd34af44ba3a442cfd03115758b79/sentry/src/main/java/io/sentry/Attachment.java#L16-L17